src/interpolator.cpp

```diff
@@ -36,6 +36,30 @@ namespace simple_mpc
     }
   }
 
+  void Interpolator::interpolateState(
+    const double delay,
+    const double timestep,
+    const std::vector<Eigen::VectorXd> & xs,
+    Eigen::Ref<Eigen::VectorXd> x_interp)
+  {
+    assert(("State is not of the right size", xs[0].size() == model_.nq + model_.nv));
+
+    // Compute the time knot corresponding to the current delay
+    size_t step_nb = static_cast<size_t>(delay / timestep);
+    double step_progress = (delay - (double)step_nb * timestep) / timestep;
+
+    // Interpolate state trajectory
+    if (step_nb >= xs.size() - 1)
+      x_interp = xs.back();
+    else
+    {
+      x_interp.head(model_.nq) =
+        pinocchio::interpolate(model_, xs[step_nb].head(model_.nq), xs[step_nb + 1].head(model_.nq), step_progress);
+      x_interp.tail(model_.nv) =
```

Comment:
I think you could have just make calsl to
```c++
interpolateConfiguration(delay, timestep, xs.head(), x_interp.head())
interpolateLinear(delay, timestep, xs.tail(), x_interp.tail())
```

Reply:
Well, the problem in this context is that xs is a std::vector<Eigen::VectorXd>, and we have to convert xs into a qs and vs vectors. This is bothersome since in Go2_MPC_experiments we directly obtained xs from the message conversion.

Reply:
Yes I realized it just now, that it would be more annoying than this. My bad